src/repository/scriptRepository.js

```diff
@@ -0,0 +1,83 @@
+const path = require('path');
+const Promise = require('bluebird');
+const fs = Promise.promisifyAll(require('node:fs'));
+
+const db = require('./database.js');
+const { Character, parseTitle, parseLines, parseCharacters } = require('../playData.js');
+
+function getCharacters(title, actorId = -1) {
+  const values = [actorId, title];
+  const sql = `
+      SELECT c.id as id, c.name as name, line_count, speaks_count
+      FROM characters c
+               JOIN scripts s ON script_id = s.id
+      WHERE ($1 = -1 or actor_id = $1)
+        AND s.title = $2`;
+
+  return db.query(sql, values).then((result) => {
+    return result.rows.map((c) => new Character(c.name, c.line_count, c.speaks_count));
+  });
+}
+
+function getTitles() {
+  return db.query('select id, title from scripts').then((result) => {
+    return result.rows.map((s) => ({
+      id: s.id,
+      title: s.title
+    }));
+  });
+}
+
+function getScript(scriptDir, title) {
+  return db.query('select filename from scripts where title = $1 limit 1', [title])
+    .then((result) => {
+      return fs.readFileAsync(scriptDir + '/' + result.rows[0].filename, 'utf8');
```

Comment:
Neat thing about putting all this in the repository class is that from the outside, we'd never know whether the script is stored on the filesystem or in the database. We could change where we save it and not need to update any other files.